src/treemap/split.js

```diff
@@ -0,0 +1,105 @@
+export default function(parent, x0, y0, x1, y1) {
+
+  parent.x0 = x0;
+  parent.x1 = x1;
+  parent.y0 = y0;
+  parent.y1 = y1;
+
+  // exit if no children
+  if (!(parent.children) || parent.children.length === 0) {
+    return;
+  }
+
+  splitTree(parent.children, parent.value, x0, y0, x1, y1);
+
+  function splitTree(nodes, sum, x0, y0, x1, y1) {
+
+    var i, n = nodes.length;
+
+    if (n === 0) {
+      return;
+    }
+
+    if (n === 1) {
+      nodes[0].x0 = x0;
+      nodes[0].x1 = x1;
+      nodes[0].y0 = y0;
+      nodes[0].y1 = y1;
+      if (nodes[0].children && nodes[0].children.length > 0) {
+        splitTree(nodes[0].children, parent.value, x0, y0, x1, y1);
```

Comment:
This looks like it assumes that *parent*.value is equal to the sum of the children’s values, which won’t be the case if the parent has a self-value.